### PR TITLE
JAMES-2279 Add WithPriority mailet

### PR DESCRIPTION
**Pull request by clitetailor**

### CODE REVIEW

File: server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithPriority.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ ****************************************************************/
+
+package org.apache.james.transport.mailets;
+
+import org.apache.james.queue.api.MailPrioritySupport;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMailet;
+
+import javax.mail.MessagingException;
+import java.util.Optional;
```
Comment (chibenwa) [inline server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/WithPriority.java]: Hi!

We do use  a specifc import layout and ordering.

In this comment I present it to another contributor: https://github.com/apache/james-project/pull/119#issuecomment-398259479

Here the ordering will be:

```
import java.util.Optional;

import javax.mail.MessagingException;

import org.apache.james.queue.api.MailPrioritySupport;
import org.apache.mailet.Mail;
import org.apache.mailet.base.GenericMailet;
```

For the record here is the IntelliJ configuration:

```
import static all other imports
<blank line>
import java.*
<blank line>
import javax.*
<blank line>
import org.*
<blank line>
import com.*
<blank line>
import all other imports
```

(intelliJ -> File -> Settings -> Editor -> Code style -> Java -> Imports -> Import layout)

BTW you can check the checkstyle by running `mvn clean install -DskipTests` then `mvn checkstyle:check`.

File: server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/WithPriorityTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import org.apache.james.queue.api.MailPrioritySupport;
+import org.apache.mailet.Mail;
+import org.apache.mailet.MailetConfig;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMailContext;
+import org.apache.mailet.base.test.FakeMailetConfig;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.*;
```
Comment (chibenwa) [inline server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/WithPriorityTest.java]: We do not use wildcard import, but do import all statements, one by one.

Also there is some import ordering issues in this class too. It should be:

```
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatCode;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.apache.james.queue.api.MailPrioritySupport;
import org.apache.mailet.Mail;
import org.apache.mailet.MailetConfig;
import org.apache.mailet.base.test.FakeMail;
import org.apache.mailet.base.test.FakeMailContext;
import org.apache.mailet.base.test.FakeMailetConfig;
import org.junit.Before;
import org.junit.Test;
```

**Comment by chibenwa** [comment]
Hi, one more comment:

We do use 4 space indetation, but your contribution uses TAB, generating this checkstyle violation:

```
[a2a6df63a568fb6eb5cda5d4676057d198564e0b] [INFO] There are 3 errors reported by Checkstyle 8.5 with checkstyle.xml ruleset.
[a2a6df63a568fb6eb5cda5d4676057d198564e0b] [ERROR] src/main/java/org/apache/james/transport/mailets/WithPriority.java:[26] (imports) CustomImportOrder: Import statement for 'javax.mail.MessagingException' is in the wrong order. Should be in the 'STANDARD_JAVA_PACKAGE' group, expecting not assigned imports on this line.
[a2a6df63a568fb6eb5cda5d4676057d198564e0b] [ERROR] src/main/java/org/apache/james/transport/mailets/WithPriority.java:[27] (imports) CustomImportOrder: Import statement for 'java.util.Optional' is in the wrong order. Should be in the 'STANDARD_JAVA_PACKAGE' group, expecting not assigned imports on this line.
[a2a6df63a568fb6eb5cda5d4676057d198564e0b] [ERROR] src/main/java/org/apache/james/transport/mailets/WithPriority.java:[60,5] (whitespace) FileTabCharacter: Line contains a tab character.
```

Could you remove TAB and use 4 spaces instead?

**Comment by clitetailor** [comment]
Hi, Benoit. 

Because today is my brother birthday and i have to buy my brother some Crayon Shinchan comic book sand i'm lazy. Thus, i will try to install IntellJ on Windows and clean up import all the way, this midnight!

Hope you have a good weekend. Cheers! 😄

**Comment by clitetailor** [comment]
It's quite take me a while to acquaint with building these things. First i run it on windows and the compiler complains about swagger dependency. Next i try it on linux and the compiler failed because it runs out of memory. I tried to make some tweaks ibut it seems not to work. So finally, i only run build and style check on server mailet only and this time only build success message.

Hope that everything will fine. And a few more words, sorry for having 'amend-ing' the old commit. Sincerely!

**Comment by chibenwa** [comment]
Hi,

This pull request has just been merged.

Thank you again for your dedication toward the James project. If you need help, want to [contribute](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20newbie%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) [documentation](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20documentation%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), new [features](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20feature%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), tests or [bugfix](https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20easyfix%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC), please don't hesitate to reach us.

One very last thing: I don't have write access to this repository, hence I can not close this issue. Wouldn't you mind doing it for me? It would avoid me annoying the Apache INFRA team with such simple concerns.

Thanks again,

Cheers,

Benoit

**Pull request closed without merging**